applications/asset_tracker_v2/src/ext_sensors/ext_sensors_bsec.c

```diff
@@ -0,0 +1,300 @@
+	const struct device *i2c_master;
+
+	/* Spinlock used to safely read out sensor readings from the BSEC library driver. */
+	struct k_spinlock sensor_read_lock;
+
+	/* Intermediate libraries used to hold the latest sampled sensor readings. */
+	double temperature_latest;
+	double humidity_latest;
+	double pressure_latest;
+	uint16_t air_quality_latest;
+
+	/* Internal BSEC thread metadata value. */
+	struct k_thread thread;
+
+	/* Buffer used to maintain the BSEC library state. */
+	uint8_t state_buffer[BSEC_MAX_STATE_BLOB_SIZE];
+	int32_t state_buffer_len;
+
+} ctx = { 0 };
+
+/* Forward declarations */
+static int settings_set(const char *key, size_t len_rd, settings_read_cb read_cb, void *cb_arg);
+SETTINGS_STATIC_HANDLER_DEFINE(bsec, SETTINGS_NAME_BSEC, NULL, settings_set, NULL, NULL);
+
+/* Private API */
+
+static int settings_set(const char *key, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	if (!key) {
+		LOG_ERR("Invalid key");
+		return -EINVAL;
+	}
+
+	LOG_DBG("Settings key: %s, size: %d", log_strdup(key), len_rd);
+
+	if (!strncmp(key, SETTINGS_KEY_STATE, strlen(SETTINGS_KEY_STATE))) {
+		if (len_rd > sizeof(ctx.state_buffer)) {
+			LOG_ERR("Setting too big: %d", len_rd);
+			return -EMSGSIZE;
+		}
+
+		ctx.state_buffer_len = read_cb(cb_arg, ctx.state_buffer, len_rd);
+		if (ctx.state_buffer_len > 0) {
+			return 0;
+		}
+
+		LOG_ERR("No settings data read");
+		return -ENODATA;
+	}
+
+	return -ENOTSUP;
+}
+
+static int enable_settings(void)
+{
+	int err;
+
+	err = settings_subsys_init();
+	if (err) {
+		LOG_ERR("settings_subsys_init, error: %d", err);
+		return err;
+	}
+
+	err = settings_load_subtree(settings_handler_bsec.name);
+	if (err) {
+		LOG_ERR("settings_load_subtree, error: %d", err);
+	}
+
+	return err;
+}
+
+static int8_t bus_write(uint8_t dev_addr, uint8_t reg_addr, uint8_t *reg_data_ptr, uint16_t len)
+{
+	uint8_t buf[len + 1];
+
+	buf[0] = reg_addr;
+	memcpy(&buf[1], reg_data_ptr, len);
+	return i2c_write(ctx.i2c_master, buf, len + 1, dev_addr);
+}
+
+static int8_t bus_read(uint8_t dev_addr, uint8_t reg_addr, uint8_t *reg_data_ptr, uint16_t len)
+{
+	return i2c_write_read(ctx.i2c_master, dev_addr, &reg_addr, 1, reg_data_ptr, len);
+}
+
+static int64_t get_timestamp_us(void)
+{
+	return k_uptime_get() * MSEC_PER_SEC;
+}
+
+static void delay_ms(uint32_t period)
+{
+	k_sleep(K_MSEC(period));
+}
+
+static void output_ready(int64_t timestamp, float iaq, uint8_t iaq_accuracy, float temperature,
+			 float humidity, float pressure, float raw_temperature, float raw_humidity,
+			 float gas, bsec_library_return_t bsec_status, float static_iaq,
+			 float co2_equivalent, float breath_voc_equivalent)
+{
+	k_spinlock_key_t key = k_spin_lock(&(ctx.sensor_read_lock));
+
+	ctx.temperature_latest = (double)temperature;
+	ctx.humidity_latest = (double)humidity;
+	ctx.pressure_latest = (double)pressure / 1000;
+	ctx.air_quality_latest = (uint16_t)iaq;
+	k_spin_unlock(&(ctx.sensor_read_lock), key);
```

Comment:
I prefer blank line between variable assignment and function calls, but appears that checkpatch doesn't care, so it can be as is

Reply:
Me too, will update.